app/assets/javascripts/discourse/app/routes/signup.js

```diff
@@ -10,7 +12,11 @@ export default class SignupRoute extends DiscourseRoute {
 
   authComplete = false;
 
-  beforeModel() {
+  beforeModel({ from }) {
+    if (from) {
```

Comment:
Applies the same "store the origin url in a cookie" behavior we do when login in, but for the signup process.